pzsd_bot/__main__.py

```diff
@@ -39,10 +54,132 @@ async def on_message(message):
     if message.author == bot.user:
         return
 
+    if match := point_pattern.search(message.content):
+        point_amount = int(match[1].replace(",", ""))
+        recipient_name = match[2]
+        recipient_id = match[3]
+
+        async with engine.connect() as conn:
+            result = await conn.execute(
+                select(pzsd_user).where(
+                    pzsd_user.c.discord_snowflake == str(message.author.id)
+                )
+            )
+            try:
+                bestower = result.one()
+            except NoResultFound:
+                logger.info(
+                    "User '%s' with snowflake '%s' tried to bestow points but wasn't in the user table",
+                    message.author.name,
+                    message.author.id,
+                )
+                return
+
+            if recipient_name:
+                result = await conn.execute(
+                    select(pzsd_user).where(pzsd_user.c.name == recipient_name.lower())
+                )
+            else:
+                result = await conn.execute(
+                    select(pzsd_user).where(
+                        pzsd_user.c.discord_snowflake == recipient_id
+                    )
+                )
+            try:
+                recipient = result.one()
+            except NoResultFound:
+                logger.info(
```

Comment:
should probably be a logger.warning

Reply:
I couldn't decide, since there not being a result isn't exactly "unexpected" so to speak. I could just as easily use like the [`result.one_or_none()`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.one_or_none) method instead which would just return `None` if there were no results and only raise an exception if multiple results came back. Honestly that is maybe more fitting here.

Reply:
gonna go with that, let me know if you still think I should handle it differently though